functional-test/src/main/java/org/zanata/page/WebDriverFactory.java

```diff
@@ -136,6 +144,29 @@ private WebDriver configureChromeDriver() {
                                 capabilities)));
     }
 
+    /**
+     * Returns a File pointing to a chromedriver binary, searching the PATH
+     * if necessary. If the property 'webdriver.chrome.driver' points to an
+     * executable, uses that.  Otherwise, searches PATH for the specified
+     * executable.  Searches for 'chromedriver' if the property is null or
+     * empty.
+     * @return a File pointing to the executable
+     * @deprecated use ChromeDriverService.createDefaultService() if you can
+     */
+    @Deprecated
+    private File getChromeDriver() {
+        String exeName = PropertiesHolder.getProperty("webdriver.chrome.driver");
+        if (exeName == null || exeName.isEmpty()) {
+            exeName = CommandLine.find("chromedriver");
+        } else if (!new File(exeName).canExecute()) {
+            exeName = CommandLine.find(exeName);
```

Comment:
Well, perhaps, but I'd rather have a deprecated function call than a
hard-coded path to a file on someone's machine.  I wasn't able to find a
way to avoid the function call without writing a lot of single-use code.

On 2014-12-22 14:24, Damian Jansen wrote:
> In functional-test/src/main/java/org/zanata/page/WebDriverFactory.java
> <https://github.com/zanata/zanata-server/pull/655#discussion-diff-22154202>:
> 
>> +    /**
>> +     * Returns a File pointing to a chromedriver binary, searching the PATH
>> +     * if necessary. If the property 'webdriver.chrome.driver' points to an
>> +     * executable, uses that.  Otherwise, searches PATH for the specified
>> +     * executable.  Searches for 'chromedriver' if the property is null or
>> +     * empty.
>> +     * @return a File pointing to the executable
>> +     * @deprecated use ChromeDriverService.createDefaultService() if you can
>> +     */
>> +    @Deprecated
>> +    private File getChromeDriver() {
>> +        String exeName = PropertiesHolder.getProperty("webdriver.chrome.driver");
>> +        if (exeName == null || exeName.isEmpty()) {
>> +            exeName = CommandLine.find("chromedriver");
>> +        } else if (!new File(exeName).canExecute()) {
>> +            exeName = CommandLine.find(exeName);
> 
> Isn't it against The Rules™
> <https://www.securecoding.cert.org/confluence/display/java/MET02-J.+Do+not+use+deprecated+or+obsolete+classes+or+methods>
> to introduce usage of deprecated functions?
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/655/files#r22154202>.
> 


-- 
Sean Flanigan

Senior Software Engineer
Engineering - Internationalisation
Red Hat